components/page.go

```diff
@@ -30,6 +30,22 @@ type Page struct {
 	Layout Layout
 }
 
+type PageOpts func(p *Page)
+
+func (p *Page) SetPageOptions(opts ...PageOpts) *Page {
+	for _, opt := range opts {
+		opt(p)
+	}
+	return p
+}
```

Comment:
Sorry to interfere like this, i forked it and i see you made the same changes. You can add Layout options ass well
```go
func WithLayoutOpts(l Layout) PageOpts {
        return func(p *Page) {
		p.Layout = l
	}
}
```